src/Cart/Cart-value.js

```diff
@@ -0,0 +1,185 @@
+import { filterDataOnStatus } from "../Product-listing";
+import { useEffect, useState } from "react";
+import { coupons } from "../database";
+import { useStateContext } from "../context";
+import "./cart.css";
+
+export const CartValueDetails = () => {
+  const { state } = useStateContext();
+  const [userSelectedCoupon, setCoupon] = useState({
+    couponName: "",
+    couponPrice: 0,
+    minOrderValue: ""
+  });
+  const [showOfferModal, setOfferModal] = useState(false);
+
+  const dataToMap = filterDataOnStatus(state.itemsInCart);
+
+  const cartDetailsCalculator = (data) =>
+    data.reduce(
+      (sum, { price, cartQty, offer }) => {
+        return {
+          totalMRP: sum.totalMRP + Number(price) * Number(cartQty),
+          discount:
+            sum.discount +
+            (Number(price) * Number(cartQty) * Number(offer)) / 100
+        };
+      },
+      { totalMRP: 0, discount: 0 }
+    );
+  const cartDetails = cartDetailsCalculator(dataToMap);
+  const cartTotalWithoutOffer = cartDetails.totalMRP - cartDetails.discount;
+  const cartTotal =
+    cartDetails.totalMRP -
+    cartDetails.discount -
+    Number(userSelectedCoupon.couponPrice);
+
+  useEffect(() => {
+    const checkOffersValid = () => {
+      if (cartTotalWithoutOffer <= Number(userSelectedCoupon.minOrderValue)) {
+        setCoupon({
+          couponName: "",
+          couponPrice: 0,
+          minOrderValue: ""
+        });
+      }
+    };
+    checkOffersValid();
+  }, [state.itemsInCart]);
+
+  return (
+    <>
+      <div className="column-40-pc padding-1rem-borderbox border-width-2px-left display-flex-column">
+        <div className="text-regular-weight body-cp-md">COUPONS</div>
+
+        <button
+          onClick={() => setOfferModal(true)}
+          style={{ textAlign: "left" }}
+          className="btn btn-text-icon-secondary btn-outline-secondary"
+        >
+          <span className="btn-icon">
+            <i className="fas fa-tag"></i>
+          </span>
+          Apply Coupon
+        </button>
+
+        <div className="p text-regular-weight body-cp-md">
+          PRICE DETAILS: ({dataToMap.length} items)
+        </div>
+
+        <div className="row body-cp-md ">
+          <div className="column-80-pc">Total MRP</div>
+          <div className="column-20-pc text-right">
+            &#8377;{cartDetails.totalMRP.toFixed(2)}
+          </div>
+        </div>
+        <div className="row body-cp-md ">
+          <div className="column-80-pc">Discount on MRP</div>
+          <div className="column-20-pc text-right text-green">
+            &#8377;{cartDetails.discount.toFixed(2)}
+          </div>
+        </div>
+        {userSelectedCoupon.couponPrice !== 0 && (
+          <div className="row body-cp-md ">
+            <div className="column-80-pc">Coupon Discount</div>
+            <div className="column-20-pc text-right text-green">
+              &#8377;{userSelectedCoupon.couponPrice}

```

Comment:
`&#8377`  - what is this i am not  aware

Reply:
This is a Unicode char for Rs. symbol, for more clarity, I will create a varibale and use it in the html